applications/tari_dan_wallet_cli/src/command/nfts.rs

```diff
@@ -0,0 +1,164 @@
+//   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//   SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//   WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//   USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{
+    fs,
+    io::{self, Read},
+    path::PathBuf,
+    str::FromStr,
+};
+
+use anyhow::anyhow;
+use clap::{Args, Subcommand};
+use tari_template_lib::{
+    crypto::RistrettoPublicKeyBytes,
+    prelude::{Amount, NonFungibleAddress},
+};
+use tari_utilities::hex::Hex;
+use tari_wallet_daemon_client::{types::MintAccountNFTRequest, ComponentAddressOrName, WalletDaemonClient};
+
+use crate::command::transaction::summarize_finalize_result;
+
+#[derive(Debug, Subcommand, Clone)]
+pub enum AccountNFTSubcommand {
+    #[clap(alias = "mint")]
+    Mint(MintAccountNFTArgs),
+}
+
+#[derive(Debug, Args, Clone)]
+pub struct MintAccountNFTArgs {
+    #[clap(long, short = 'a', alias = "account")]
+    pub account: Option<ComponentAddressOrName>,
+    #[clap(long, short = 'o', alias = "owner-token")]
+    pub owner_token: String,
+    #[clap(long, short = 't', alias = "token-symbol")]
+    pub token_symbol: Option<String>,
+    #[clap(long, short = 'i', alias = "metadata-file")]
+    pub metadata_file: Option<PathBuf>,
+    #[clap(long, short = 'm', alias = "metadata")]
+    pub metadata: Option<serde_json::Value>,
+    #[clap(long, short = 'f', alias = "mint-fee")]
+    pub mint_fee: Option<u32>,
+    #[clap(long, short = 'c', alias = "create-account-nft-fee")]
+    pub create_account_nft_fee: Option<u32>,
+}
+
+impl AccountNFTSubcommand {
+    pub async fn handle(self, mut client: WalletDaemonClient) -> Result<(), anyhow::Error> {
+        match self {
+            Self::Mint(args) => {
+                handle_mint_account_nft(args, &mut client).await?;
+            },
+        }
+        Ok(())
+    }
+}
+
+pub async fn handle_mint_account_nft(
+    args: MintAccountNFTArgs,
+    client: &mut WalletDaemonClient,
+) -> Result<(), anyhow::Error> {
+    let MintAccountNFTArgs {
+        account,
+        owner_token,
+        token_symbol,
+        metadata_file,
+        metadata,
+        mint_fee,
+        create_account_nft_fee,
+    } = args;
+
+    let account = if let Some(account) = account {
+        account
+    } else {
+        println!(
+            "Please paste console wallet account name or respective component address from mint_account_nft call in \
+             the terminal: Press <Ctrl/Cmd + d> once done"
+        );
+
+        let mut account = String::new();
+        io::stdin().read_to_string(&mut account)?;
+        ComponentAddressOrName::from_str(&account)
+            .map_err(|e| anyhow!("Failed to parse account name or component address, with error = {}", e))?
```

Comment:
Since the input is a single line, can use `read_line` - then the user just has to press enter

```suggestion
        println!(
            "Please paste console wallet account name or respective component address from mint_account_nft call in \
             the terminal
        );

        let mut account = String::new();
        io::stdin().read_line(&mut account)?;
        ComponentAddressOrName::from_str(&account)
            .map_err(|e| anyhow!("Failed to parse account name or component address, with error = {}", e))?
```

Reply:
Great suggestion !